docket/config.py

```diff
@@ -0,0 +1,153 @@
+        'CELERY_TASK_IGNORE_RESULT': True,
+        'LOG_MSG_FORMAT': '[%(processName)-8s] %(message)s',
+        'LOG_DATE_FORMAT': '%Y-%m-%dT%H:%M:%SZ',
+        'LOG_DATE_UTC': True,
+        'LOG_LEVEL': 'info',
+        'LOG_FILE_BACKUPS' : 7,
+        'LOG_FILE_MSG_FORMAT': '%(asctime)s[%(processName)-8s] %(message)s',
+        'LOG_FILE_DATE_FORMAT': '%Y%jT%H:%M:%S',
+        'LOG_FILE_LEVEL': 'info',
+    }
+    _default = config.copy()
+
+    @classmethod
+    def load(cls, filename=None, flask_app=None, logger=None):
+        """ Set Config from a yaml file:
+            filename a config file: search for the file in several directories
+        """
+
+        import itertools
+        app_root = flask_app.root_path if flask_app else '/opt/docket/docket'
+        fs_root  = os.path.abspath(os.sep)
+        paths = itertools.product(
+            ( filename,                     # filename.yaml
+             'docket',                      # docket.yaml
+            ),
+            [ tuple(),                      # <empty> - filename can be any valid path
+             (fs_root, 'etc', 'docket'),    # /etc/docket/filename
+             (fs_root, 'etc',),             # /etc/filename
+#            (os.curdir, os.pardir, 'conf'),# CWD/../conf/filename
+             (app_root, 'conf'),            # APP/conf/filename
+             ('conf',)]                     # ./conf/filename
+        )
+        for f,p in paths:
+            if not f:
+                continue
+            filepath = os.path.join(*(p + (f,)) )
+            filepath = os.path.expanduser(filepath)
+            if os.path.isfile(filepath):
+                break
+            elif os.path.isfile(filepath+'.yaml'):
+                filepath += '.yaml'
+                break
+            print("Config: >{}< not found".format(filepath))
+
+        # load the config file and update our class config data
+        with open(filepath) as f:
+            cls.config.update(yaml.load(f.read()))
+        print("Config: loaded >{}<".format(filepath))
+
+        # I'm having the hardest time getting a logging to work
+        cls.loggers['docket']= logging.getLogger(Config.get('LOGGER_NAME','docket'))
+        if isinstance(logger, logging.Logger):
+            cls.loggers[None] = logger
+        if flask_app:
+            flask_app.config.from_mapping(cls.config)
+            cls.loggers['flask'] = flask_app.logger
+            #cls.config = flask_app.config
+
+        cls._setup_loggers()
+
+        return cls.config
+
+    @classmethod
+    def _setup_loggers(cls):
+        cls.log_level = log_level(cls.config['LOG_LEVEL'])
+        cls.formatter = logging.Formatter(fmt=cls.config['LOG_MSG_FORMAT'],
+                                          datefmt=cls.config['LOG_DATE_FORMAT'])
+
+        handler = None
+        if not cls.file_handler and cls.get('LOG_FILE', None):
+            from logging.handlers import TimedRotatingFileHandler
```

Comment:
I'd be inclined to default to journald output (I think syslog interface will get it there), or let `logrotate` do its thing.

Reply:
actually...looks like systemd has its own logging handler. https://stackoverflow.com/a/34589151

Reply:
Optionally...I can add this after merge myself.

Reply:
Honestly the easiest way I've found to log under systemd is print().  
the various FileHandler logging is... incomplete. For example: I haven't been able to to get exceptions in that log - those are getting redirected... I think CELERY is doing some stuff.
https://stackoverflow.com/questions/10390537/cant-redirect-stdout-stderr-to-another-logger-in-celery

We prefer systemd logging anyway, and I already want to leave celery for a simpler component.  At that point I believe I can fix the remaining problems with the 'LOG_FILE' option.